tests/writer_test.py

```diff
@@ -1,39 +1,55 @@
 import os
-import unittest
-
 import xml.etree.ElementTree as ET
 
-from pyuntl.untldoc import untlxml2py
-
-
-class WriterTest(unittest.TestCase):
-    def setUp(self):
-        self.well_formed_record = untlxml2py(
-            os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                         'metadc_complete.untl.xml')
-        )
+import pytest
 
-    def test_write_xml_file_from_pyuntl(self):
-        self.well_formed_record.create_xml_file('/tmp/xml_test_output.xml')
-        self.written_record = ET.parse('/tmp/xml_test_output.xml')
-        self.well_formed_record = ET.parse(
-            os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                         'metadc_complete.untl.xml')
-        )
-        # Get canonical version of XML files.
-        cxml1 = self.well_formed_record
-        cxml2 = self.written_record
-        # Compare initial XML to generated XML.
-        self.assertTrue(ET.tostring(cxml1.getroot()) == ET.tostring(cxml2.getroot()))
-
-    def tearDown(self):
-        os.remove('/tmp/xml_test_output.xml')
-
-
-def suite():
-    test_suite = unittest.makeSuite(WriterTest, 'test')
-    return test_suite
+from pyuntl.untldoc import untlxml2py
 
 
-if __name__ == '__main__':
-    unittest.main()
+@pytest.mark.parametrize('original_file', ['metadc_complete.untl.xml',
+                                           'metadc_utf8.untl.xml'])
+def test_write_xml_file_from_pyuntl_is_identical(original_file, tmpdir):
+    # Load data from XML file into a UNTLElement tree.
+    original_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                 original_file)
+    well_formed_record = untlxml2py(original_path)
+    # Write the loaded data back to a new XML file.
+    new_path = os.path.join(tmpdir, 'xml_test_output.xml')
+    well_formed_record.create_xml_file(new_path)
+    # Read back the new file and original files into ElementTrees.
```

Comment:
I'm a little confused as to why the next few lines are needed if at the end of this test we check the equality of the written files directly.

Reply:
Similar question about the next test too.

Reply:
I don't remember why I did it that way--maybe I left it in there like that only because that is how it was originally tested in the `unittest` version? I will remove the redundancy.

Reply:
Hmm I guess in the next (ASCII) test it shows that in ElementTree bytes form they are equal, even though on disk the files are different.